integration/commands_administration_test.go

```diff
@@ -773,6 +775,227 @@ func TestCommandsAdministrationDBStatsEmptyWithScale(t *testing.T) {
 	// https://github.com/FerretDB/FerretDB/issues/727
 }
 
+func TestCommandsAdministrationRenameCollection(t *testing.T) {
+	t.Parallel()
+
+	maxTableLen := strings.Repeat("a", 63) // maxTableLen = 63
+	insertCollections := []string{"foo", "buz"}
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		collection       string
+		targetCollection string
+		expected         bson.D
+		err              *mongo.CommandError
+		toInsert         []interface{}
+	}{
+		"Rename": {
+			collection:       "foo",
+			targetCollection: "bar",
+			expected:         bson.D{{"ok", float64(1)}},
+		},
+		"RenameSame": {
+			collection:       "foo",
+			targetCollection: "foo",
+			err: &mongo.CommandError{
+				Code:    20,
+				Name:    "IllegalOperation",
+				Message: `Can't rename a collection to itself`,
+			},
+		},
+		"RenameDuplicate": {
+			collection:       "foo",
+			targetCollection: "buz",
+			err: &mongo.CommandError{
+				Code:    48,
+				Name:    "NamespaceExists",
+				Message: `target namespace exists`,
+			},
+		},
+		"SoureDoesNotExist": {
+			collection:       "none",
+			targetCollection: "bar",
+			err: &mongo.CommandError{
+				Code:    26,
+				Name:    "NamespaceNotFound",
+				Message: "source namespace does not exist",
+			},
+		},
+		// this confirms that after we rename foo to bar and then create foo again,
+		// 1. bool-false doesn't exist
```

Comment:
Isn't it easier to test this condition by inserting `{"_id":"bool-false"}`. Because it should work if it doesn't exist.